src/player.h

```diff
@@ -76,6 +77,8 @@ enum edible_rating {
     NO_TOOL
 };
 
+using edible_ret_val = ret_val<edible_rating, EDIBLE, INEDIBLE>;
```

Comment:
This isn't very intuitive, needs comments. Preferably in `ret_val` structure.
First template argument is obvious. From context, it looks like the second one is either default value or "value that means `true`", with the third also being possible default or "value that means `false`".